File: app/src/main/java/com/simplecity/amp_library/dagger/module/ActivityModule.java
package com.simplecity.amp_library.dagger.module;

import android.app.Activity;
import com.simplecity.amp_library.dagger.scope.ActivityScope;
import com.simplecity.amp_library.playback.MediaManager;
import com.simplecity.amp_library.ui.activities.BaseActivity;
import dagger.Module;
import dagger.Provides;

@Module
public class ActivityModule {

    private Activity activity;

    public ActivityModule(Activity activity) {
        this.activity = activity;
    }

    @Provides
    @ActivityScope
    Activity provideActivity() {
        return activity;
    }

    @Provides
    @ActivityScope
    MediaManager provideMediaManager() {
        return ((BaseActivity) activity).getMusicUtils();
    }
}

File: app/src/main/java/com/simplecity/amp_library/di/app/activity/ActivityModule.kt
package com.simplecity.amp_library.di.app.activity

import android.app.Activity
import androidx.fragment.app.FragmentManager
import androidx.appcompat.app.AppCompatActivity
import dagger.Module
import dagger.Provides

@Module
class ActivityModule {

    @Provides
    @ActivityScope
    fun activity(activity: AppCompatActivity): Activity {
        return activity
    }

    @Provides
    @ActivityScope
    fun fragmentManager(activity: AppCompatActivity): FragmentManager {
        return activity.supportFragmentManager
    }
}